agent/command/git.go

```diff
@@ -352,14 +352,19 @@ func (c *gitFetchProject) waitForMergeableCheck(ctx context.Context, comm client
 	)
 	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
 	attempt := 0
+	mergeableCheckErr := false
 	err := utility.Retry(ctx, func() (bool, error) {
+		mergeableCheckErr = false
 		attempt++
 		lastAttempt := attempt == getPRAttempts
 		info, err := comm.GetPullRequestInfo(ctx, td, conf.GithubPatchData.PRNumber, opts.owner, opts.repo, lastAttempt)
 		if err != nil {
 			return false, errors.Wrap(err, "getting pull request data from GitHub")
 		}
 		if info.Mergeable == nil {
+			// TODO EVG-19723: if using the merge commit SHA here doesn't cause issues, we should remove retrying here.
+			mergeSHA = info.MergeCommitSHA
+			mergeableCheckErr = true
```

Comment:
I think it's easier to understand if we prepend `if mergeSHA != "" { return true, nil }` right before L368 and then remove the change at the end of the function.

Reply:
If we don't return an error, then the function won't retry. This should theoretically also be temporary so I don't know how much we need to optimize.

Reply:
Gotcha, thanks